src/core/thread/router_table.hpp

```diff
@@ -342,7 +342,7 @@ class RouterTable : public InstanceLocator
     }
 
     Router   mRouters[Mle::kMaxRouters];
-    uint8_t  mAllocatedRouterIds[BitVectorBytes(Mle::kMaxRouterId)];
+    uint8_t  mAllocatedRouterIds[BitVectorBytes(Mle::kMaxRouterId + 1)];
```

Comment:
Router Ids can be from `0` up to (including) `Mle::kMaxRouterId` (which requires `Mle::kMaxRouterId + 1` bits)
I think `BitVectorBytes(x)` returns number of bytes to store bit-vector capable of storing `x` bits. 
With default value of ` Mle::kMaxRouterI = 62` it should not matter though.

Reply:
Good catch. However, I wouldn't label this as a bug since this does not actually change any visible behavior. Note that `Mle::kMaxRouterId` is not a configurable value and is specified by the Thread specification.

Reply:
Sounds good. I was not sure about "bug" label either.